NEXTCLOUD.cmake

```diff
@@ -15,6 +15,34 @@ set( APPLICATION_VIRTUALFILE_SUFFIX "nextcloud" CACHE STRING "Virtual file suffi
 set( LINUX_PACKAGE_SHORTNAME "nextcloud" )
 set( LINUX_APPLICATION_ID "${APPLICATION_REV_DOMAIN}.${LINUX_PACKAGE_SHORTNAME}")
 
+option(DEV_BUILD "If enabled, name variables will be appended with DEV_SUFFIX in order to reduce conflicts." OFF)
+if (NOT DEV_SUFFIX)
+  set(DEV_SUFFIX "Dev"
+    CACHE STRING
+      "Choose a suffix with which to append build name variables to prevent conflicts."
+  FORCE)
+endif ()
+
+function(ADD_SUFFIXES SUFFIX_STRING)
+    string(TOLOWER ${SUFFIX_STRING} SUFFIX_LOWER)
+    string(APPEND APPLICATION_NAME " (${SUFFIX_STRING} Build)")
+    set(APPLICATION_NAME ${APPLICATION_NAME} PARENT_SCOPE)
+    string(APPEND APPLICATION_SHORTNAME "${SUFFIX_STRING}")
+    set(APPLICATION_SHORTNAME ${APPLICATION_SHORTNAME} PARENT_SCOPE)
+    string(APPEND APPLICATION_EXECUTABLE "${SUFFIX_LOWER}")
```

Comment:
I don't know if I'm a fan of forcing this to be lower case because that somehow assumes `APPLICAION_EXECUTABLE` is always in lower case. For simplicity I would just append `SUFFIX_STRING`:) In the future we might want to have discussion if we really need different variables for `APPLICATION_NAME` and `APPLICATION_EXECUTABLE`.

Reply:
I don't think it *hurts* to force it to lowercase for the lowercase variables. It's just more a e s t h e t i c.

Reply:
I feel like it's better to work with the way things currently are than to hold out for a later decision on `APPLICATION_EXECUTABLE` not necessarily being lowercase. Remember that this code currently only applies to the "Nextcloud" brand, so generalizing this specific behavior to allow for a non-lowercase `APPLICATION_EXECUTABLE` could happen down the line if or when the suffix code gets generalized for OEM branding in the first place.